### ISSUE TITLE
Remove underscores in root-level ontology terms

### ISSUE THREAD

**Issue by ukemi** [geneontology/go_announcements#24]
There is a proposal to remove the underscores from the root-level terms in the ontology. Will this cause any major problems? Hopefully people are using identifiers.

molecular_function -> molecular function
biological_process -> biological process
cellular_component -> cellular component

https://github.com/geneontology/go-ontology/commit/6fb7e0e8a82ab003832db931fd001496f46d6f5c

**Comment by srengel** [comment]
Hi David,
what is the expected timeframe for this change?

**Comment by ukemi** [comment]
Depends on how people respond. If people have the strings hard-coded, then we need to give some lead time. It is similar to any other term name change, but these are special since they are the root nodes, so I thought it best to check.

**Comment by tberardini** [comment]
I was able to dig up the last time this was proposed (June 2009) and a slew of objections at the time.  Is it worth pasting that email thread here?

**Comment by ukemi** [comment]
Yes please.   @balhoff

**Comment by tberardini** [comment]
Date: Thu, 04 Jun 2009 08:31:29 -0400

**Comment by tberardini** [comment]
RGD (back then):
From: Stoddard, Alexander
Sent: Thursday, June 04, 2009 9:40 AM
To: Petri, Victoria; Bakir, Burcu; De Pons, Jeff
Cc: Shimoyama, Mary; Patzer, Andrew
Subject: RE: [Gofriends] Alert: proposal to remove underscores from rootterm names

A quick check of the code base shows that it does depend on the term name (NOT the stable GO id) to look up the root terms.
In the ontology loading code the file "createDBMFiles.pl" uses a search for the terms "biological_process" etc to look up the root.

At least that file will need to be updated or our ontology stuff will break.

I have not checked the website code yet, just the ontology loading code.

Our ontology updates are currently on a manual schedule, not an automatic update, so I can coordinate testing of our ontology / Portal code with the change without the risk of the public site breaking.

Given the timing of the change (I may be the only RGD developer here) and the possibility that it won't happen at all I propose waiting to see if it actually happens. If so, I can then spend time to patch our code and test it.


Thank you,
Alex



-----Original Message-----
From: Petri, Victoria
Sent: Thursday, June 04, 2009 8:53 AM
To: Stoddard, Alexander; Bakir, Burcu; De Pons, Jeff
Subject: FW: [Gofriends] Alert: proposal to remove underscores from rootterm names

Hello,

I don't know who is going to take care of uploading ontologies, but here is a proposal for changing the root terms of the three GO ontologies from underscore to space (and reaction to it). I know that in the past when such a change was introduced - I believe it was the opposite - we had problems loading the ontologies.

Please let me know and/or reply to this forwarded message.

Thanks,

Victoria

**Comment by tberardini** [comment]
from: | Benjamin Hitz <hitz@genome.stanford.edu>
-- | --
to: | "Petri, Victoria" <vpetri@mcw.edu>
cc: | GO Friends <gofriends@genome.stanford.edu>
date: | Jun 4, 2009, 10:07 AM


I object too as I am sure we have this hard coded somewhere.  I agree 
with the sentiment, however.
It's a special case - but the root terms ARE special cases (especially 
since they now essentially mean "unknown")


Ben

**Comment by tberardini** [comment]
hjd@informatics.jax.org via genome.stanford.edu | Jun 4, 2009, 10:18 AM
Harold Drabkin hjd@informatics.jax.org via genome.stanford.edu

Our SE's took a look at a test file and yes it will brake a lot of
stuff, so we object to the change also.

hjd

**Comment by tberardini** [comment]
Mark Wilkinson
+1

Perhaps just change the rdf:label if people are finding the underscore 
objectionable...??


Mark

**Comment by balhoff** [comment]
Is nine years enough time? :smile:

Interesting to see reactions from last time.

**Comment by tberardini** [comment]
Sorin Draghici: 
 Jun 4, 2009, 1:53 PM





Midori,

The Onto-Tools should be ok with the proposed changes. I bet that
somebody, somewhere did hard code these :-), but it should not be a
big deal to fix the source code since we know exactly what to look for.

However, if you decide to go ahead with it, I propose to make this
change together with other necessary changes, next time GO needs a
major overhaul. The overhead of switching somebody from whatever they
are doing to fixing this is not justified for this issue alone.

All the best,

Sorin Draghici

**Comment by tberardini** [comment]
OK, that's all of the thread. I suspect that some places will still have the same code in place that would still break, but we'll see!

**Comment by ukemi** [comment]
Thanks @tberardini

**Comment by pgaudet** [comment]
out of date
